### PR TITLE
fix outcome printer bugs

### PR DESCRIPTION
**Pull request by glennsl**
I took the existing expected test result, pasted it into the rescript playground, fixed all the errors there, then fixed the code to actually produce that result. A pretty rough process, but the fixes were all pretty simple so I hope this is accurate enough.

### CODE REVIEW

**Comment by cristianoc** [comment]
This is great, we can use this immediately in the editor extension (which vendors the outcome printer).
In particular, the arity 0 bug is very visible there on hover.

For the alias type, would you add this example:
```rescript
let alias1: [> rgb] as 'a
let alias2: ([> rgb] as 'a)
```
For the deciding when to add the parens, there should be same analogous logic in the normal pretty printer for signatures.

File: src/res_outcome_printer.ml [hunk 1]
```diff
@@ -233,17 +233,19 @@ let printPolyVarIdent txt =
...
+         Doc.rparen
        ]
      | Otyp_constr (
         Oide_dot (Oide_dot (Oide_ident "Js", "Fn") , "arity0"), (* Js.Fn.arity0 *)
         [Otyp_constr (Oide_ident ident, [])] (* int or unit or string *)
        ) ->
         (* Js.Fn.arity0<int> -> (.) => int*)
         Doc.concat [
-          Doc.text "(.) => ";
+          Doc.text "(. ()) => ";
```
Comment (IwanKaramazow) [inline src/res_outcome_printer.ml:248]: I wonder if we should also support `(.) => int` when parsing type expressions. We do support it in non-type expressions.

Comment (cristianoc) [reply src/res_outcome_printer.ml:248]: I did not know it was not supported. Sounds like it should be, to avoid confusion.

Comment (glennsl) [reply src/res_outcome_printer.ml:248]: > For the alias type, would you add this example:
> 
> ```rescript
> let alias1: [> rgb] as 'a
> let alias2: ([> rgb] as 'a)
> ```

Not entirely sure what you mean here. I can add something like this to `test/oprint/oprint.res`:

```rescript
let alias1: [> rgb] as 'a = #Red
let alias2: ([> rgb] as 'a) = #Green
```

but that wouldn't cause the outcome printer to produce an alias, just `[> rgb]`.

**Comment by glennsl** [comment]
> For the alias type, would you add this example:
> 
> ```rescript
> let alias1: [> rgb] as 'a
> let alias2: ([> rgb] as 'a)
> ```

Not entirely sure what you mean here. I can add something like this to `test/oprint/oprint.res`:

```rescript
let alias1: [> rgb] as 'a = #Red
let alias2: ([> rgb] as 'a) = #Green
```

but that wouldn't cause the outcome printer to produce an alias, just `[> rgb]`.

File: src/res_outcome_printer.ml (hunk 1 above)
Comment (glennsl) [reply src/res_outcome_printer.ml:248]: This surprised me too, and I think it should.

But also, for the record, `refmt` will convert `(.)` into `(. ())` for value expressions and `(. ())` into `(. unit)` for type expressions. Perhaps we should run this through `refmt` to ensure it produces idiomatic code too?

**Comment by cristianoc** [comment]
@glennsl thoughts about adding parsing for `(.) => int` ?

**Comment by cristianoc** [comment]
Also, would you rebase on master.

**Comment by glennsl** [comment]
> @glennsl thoughts about adding parsing for e.g. (.) => int ?

You mean beyond agreeing that it's a good idea?

> Also, would you rebase on master.

Done, but seems I can no longer run tests on Linux?

**Comment by cristianoc** [comment]
> > @glennsl thoughts about adding parsing for e.g. (.) => int ?
> 
> You mean beyond agreeing that it's a good idea?
> 
> > Also, would you rebase on master.
> 
> Done, but seems I can no longer run tests on Linux? (Edit: I can run the commands manually though, just skipping reanalyze)

opam install reanalyze, does it work for you?

**Comment by cristianoc** [comment]
Added an issue to track parsing `(.) => ...` in type expressions add https://github.com/rescript-lang/syntax/issues/549

I guess this can be merged.

**Comment by glennsl** [comment]
> opam install reanalyze, does it work for you?

Installing it works, but running it does not. Some kind of CMI parse error being raised I think. The README says "tests will currently run on macOS only", so I figured this was known.

**Comment by cknitt** [comment]
The README is outdated, I will update it. Current status is that all tests are working on all platforms, except for roundtrip tests on Windows.

**Pull request merged**